src/sync/mutex.rs

```diff
@@ -219,8 +304,9 @@ impl<T> Mutex<T> {
     /// #
     /// # })
     /// ```
+    #[inline]
```

Comment:
In general, forced inlining of trivial operations is considered an antipattern. Do we run into issues that these methods are not inlined?

Reply:
#[inline] is not forcing inline (different from #[inline(always)]; it is just hinting the compiler that this should probably be inlined.

Usually it's a good idea to have the #[inline] notation if the method is going to be inlined across the crate boundary. In the case Mutex is polymorphic over T, so we may get away with it, but a monomorphic function will never be inlined across crate boundary because the compiler cannot see the underlying implementation.

Reply:
Correct, I misspoke there a little.

That's correct as a hint, this is only in the absence of lto, though. Considering that libstd uses the same annotations, we should follow that practice.